acme/acme/client.py

```diff
@@ -624,7 +625,31 @@ def _send_request(self, method, url, *args, **kwargs):
         kwargs.setdefault('headers', {})
         kwargs['headers'].setdefault('User-Agent', self.user_agent)
         kwargs.setdefault('timeout', self._default_timeout)
-        response = self.session.request(method, url, *args, **kwargs)
+        try:
+            response = self.session.request(method, url, *args, **kwargs)
+        except requests.exceptions.RequestException as e:
+            #pylint: disable=pointless-string-statement
+            """Requests response parsing
+
+            The requests library emits exceptions with a lot of extra text.
+            We parse them with a regexp to raise a more readable exceptions.
+
+            Example:
+            HTTPSConnectionPool(host='acme-v01.api.letsencrypt.org',
+            port=443): Max retries exceeded with url: /directory
+            (Caused by NewConnectionError('
+            <requests.packages.urllib3.connection.VerifiedHTTPSConnection
+            object at 0x108356c50>: Failed to establish a new connection:
+            [Errno 65] No route to host',))"""
+
+            err_regex = r".*host='(\S*)'.*url\: (\/\w*).*(\[Errno \d+\])([A-Za-z ]*)"
+            err_str = str(e.message)
+            if err_str is not None:
```

Comment:
I don't think you need to check if err_str is None - it should always be present in an exception. I meant you need to check the result of the re.match:

```
m = re.match(err_regex, err_tr)
if m is None:
 raise
else:
  host, path, err_no, err_msg = m.groups()
```

Also, you don't need the `res` object you created here.